cpp/locality/NeighborQuery.h

```diff
@@ -167,6 +169,10 @@ class NeighborQuery
         {
             if (args.num_neighbors == QueryArgs::DEFAULT_NUM_NEIGHBORS)
                 throw std::runtime_error("You must set num_neighbors in the query arguments when performing number of neighbor queries.");
+            if (args.r_max == QueryArgs::DEFAULT_R_MAX)
```

Comment:
Seems like we're missing a check that `scale > 1` in the `args.mode == QueryArgs::nearest` section here.

Reply:
Discussed offline, we want to keep this check in `AABBQuery`. However, the most consistent thing to do here would be to also add explicit checks that users aren't passing in arguments that are unused and therefore implicitly ignored (e.g. passing `r_guess` or `scale` to `LinkCell.query`).